runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -247,45 +250,74 @@ MM_VLHGCAccessBarrier::postStoreClassToClassLoader(J9VMThread *vmThread, J9Class
 	}
 }
 
+void
+MM_VLHGCAccessBarrier::copyArrayCritical(J9VMThread* vmThread, GC_ArrayObjectModel *indexableObjectModel,
+					J9InternalVMFunctions *functions, void **data,
+					J9IndexableObject *arrayObject, jboolean *isCopy)
+{
+	I_32 sizeInElements = (I_32)indexableObjectModel->getSizeInElements(arrayObject);
+	UDATA sizeInBytes = indexableObjectModel->getDataSizeInBytes(arrayObject);
+	*data = functions->jniArrayAllocateMemoryFromThread(vmThread, sizeInBytes);
+	if (NULL == *data) {
+		functions->setNativeOutOfMemoryError(vmThread, 0, 0);	// better error message here?
+	} else {
+		indexableObjectModel->memcpyFromArray(*data, arrayObject, 0, sizeInElements);
+		if (NULL != isCopy) {
+			*isCopy = JNI_TRUE;
+		}
+	}
+	vmThread->jniCriticalCopyCount += 1;
```

Comment:
someone already commented it, but it got lost. it's probably more correct to just count successful copies